nn_modules/decoder/decoder.py

```diff
@@ -0,0 +1,31 @@
+from torch import nn
+from torch import Tensor
+from typing import Optional
+
+
+class TopDecoder(nn.Module):
+    def __init__(self, num_layers: int, d_model: int, nhead: int) -> None:
+        super(TopDecoder, self).__init__()
+        self.layers = nn.ModuleList(
+            [nn.TransformerDecoderLayer(d_model, nhead) for _ in range(num_layers)])
+
+    def forward(self, x: Tensor, memory: Tensor) -> Tensor:
+        """Pass the inputs (and mask) through the decoder layer.
```

Comment:
we don't have mask anymore so remove it from the docstring, too.

Reply:
Done